docs/content/advanced/backend/events/events-architecture.md

```diff
@@ -0,0 +1,85 @@
+The events system in Medusa is built on a publish/subscribe architecture. Events are published by the core when certain actions take place. Those events can be subscribed to, which means, a reactive action is performed asynchronously. It is useful in a variety of scenarios including but not limited to processing heavy jobs in the background and reacting to specific events in a custom project or plugin. 
```

Comment:
@adrien2p @srindom I took the liberty of adding documentation for our events architecture to provide developers with an understanding of the retry mechanism. Feel free to give it a read. 

If there is anything that stands out, let me know, but I don't think we should nitpick on sentences and the likes. 

cc. @shahednasser

Reply:
I aim to merge tomorrow around noon.